src/k8s/cmd/k8s/k8s_join_cluster.go

```diff
@@ -62,8 +63,18 @@ func newJoinClusterCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 				return
 			}
 
+			var joinClusterConfig string
+			if opts.configFile != "" {
+				joinClusterConfig, err = getJoinConfigContents(opts.configFile)
```

Comment:
no need for an extra util function, we don't do much else here than just:

```suggestion
				b, err := os.ReadFile(opts.configFile)
				joinClusterConfig = string(b)
```